app/uk/gov/hmrc/cdsreimbursementclaimfrontend/controllers/claims/EnterMovementReferenceNumberController.scala

```diff
@@ -285,70 +285,100 @@ class EnterMovementReferenceNumberController @Inject() (
+                        case _                       => true
+                      }
+                    }
+                    .getOrElse(true)
+
+                  entryNumbersMatch match {
+                    case true  =>
+                      val updatedAnswers = answers.fold(
+                        _ =>
+                          CompleteMovementReferenceNumberAnswer(
+                            Left(entryNumber)
+                          ),
+                        complete => complete.copy(movementReferenceNumber = Left(entryNumber))
+                      )
+                      val newDraftClaim  =
+                        DraftC285Claim.newDraftC285Claim.copy(movementReferenceNumberAnswer = Some(updatedAnswers))
 
-                  val updatedJourney = fillingOutClaim.copy(draftClaim = newDraftClaim)
+                      val updatedJourney = fillingOutClaim.copy(draftClaim = newDraftClaim)
 
-                  val result = EitherT
-                    .liftF(updateSession(sessionStore, request)(_.copy(journeyStatus = Some(updatedJourney))))
-                    .leftMap((_: Unit) => Error("could not update session"))
+                      val result = EitherT
+                        .liftF(updateSession(sessionStore, request)(_.copy(journeyStatus = Some(updatedJourney))))
+                        .leftMap((_: Unit) => Error("could not update session"))
 
-                  result.fold(
-                    e => {
-                      logger.warn("could not capture change in entry number", e)
-                      errorHandler.errorResult()
-                    },
-                    _ => Redirect(routes.EnterDeclarationDetailsController.enterDeclarationDetails())
-                  )
-                case Right(mrn)        =>
-                  val updatedAnswers: CompleteMovementReferenceNumberAnswer = answers.fold(
-                    _ => CompleteMovementReferenceNumberAnswer(Right(mrn)),
-                    complete => complete.copy(movementReferenceNumber = Right(mrn))
-                  )
-                  val newDraftClaim                                         =
-                    DraftC285Claim.newDraftC285Claim.copy(movementReferenceNumberAnswer = Some(updatedAnswers))
+                      result.fold(
+                        e => {
+                          logger.warn("could not capture change in entry number", e)
+                          errorHandler.errorResult()
+                        },
+                        _ => Redirect(routes.EnterDeclarationDetailsController.enterDeclarationDetails())
+                      )
+                    case false =>
+                      Redirect(routes.CheckYourAnswersAndSubmitController.checkAllAnswers())
+                  }
+
+                case Right(mrn) =>
+                  val mrnNumbersMatch = fillingOutClaim.draftClaim.movementReferenceNumber
```

Comment:
I don't see the point of almost duplicating the code here.
We could decide beforehand if they match or not (weather it's an entry number or and MRN number)